From b8894ebca76ecc1e883ef94cc9ce9798946daaee Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 16 Jul 2024 14:50:51 +0200
Subject: [PATCH 1160/1695] drivers: clk: clk_get_rates_array() returns ordered
 rates

Explicitly state in clk_get_rates_array() inline description comment
that the output rates arrays is ordered by increasing frequency
values. This change allows to better fit the sole consumer of this
API function that is the SCMI server implementation. SCMI specification
states that discrete clock rates list shall follow this order.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5818ed9b282ebb254d54e2258f1c588e5161a549 drivers: clk: clk_get_rates_array() returns ordered rates]
Change-Id: I605be5b6294fedfcf9ed0a21be61781696d67aae
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/393545
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/include/drivers/clk.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/include/drivers/clk.h b/core/include/drivers/clk.h
index c8f7eb558..c73b4db19 100644
--- a/core/include/drivers/clk.h
+++ b/core/include/drivers/clk.h
@@ -86,7 +86,7 @@ struct clk_duty {
  * @get_parent: Get the current parent index of the clock
  * @set_rate: Set the clock rate
  * @get_rate: Get the clock rate
- * @get_rates_array: Get the supported clock rates as array
+ * @get_rates_array: Get the supported clock rates as array in increasing order
  * @get_rates_steps: Get support clock rates by min/max/step representation
  * @determine_rate: Given a target rate as input, returns the closest rate
  *		actually supported by the clock, and optionally the parent clock
@@ -247,7 +247,7 @@ TEE_Result clk_get_parent_idx(struct clk *clk, struct clk *parent,
 TEE_Result clk_set_parent(struct clk *clk, struct clk *parent);
 
 /**
- * clk_get_rates_array - Get supported rates as an array
+ * clk_get_rates_array - Get supported rates as an increasing frequencies array
  *
  * @clk: Clock for which the rates are requested
  * @start_index: start index of requested rates
-- 
2.47.2

